cpp/arcticdb/version/version_store_api.hpp

```diff
@@ -344,6 +344,8 @@ class PythonVersionStore : public LocalVersionedEngine {
         bool prune_previous_versions);
 
     void delete_snapshot_sync(const SnapshotId& snap_name, const VariantKey& snap_key);
+
+    [[nodiscard]] static std::lock_guard<std::mutex> ensure_single_thread_cpp_pybind_entry();
```

Comment:
I'm confused, I can't see where this is defined, or used?

Reply:
It is used as `py::call_guard<SingleThreadMutexHolder>()` while defining `pybind` function. 
```
m.def("foo", foo, py::call_guard<T>());
```
is equal to 
```
m.def("foo", [](args...) {
    T scope_guard;
    return foo(args...); // forwarded arguments
});
```

Reply:
Oops I see your point now. Removing